### PR TITLE
feat: export a non-gatsby remark plugin

### LINKED ISSUE #136: Support embed outside gatsby using remark plugins
Wanted to gauge the interest on making this available outside Gatsby ecosystem. I'm working on a nextjs blog where I would love to have this automatic embeds, and I think its pretty straightforward with the way the gatsby plugin works.

For distribution I think on two scenarios, the first would be to make this a multi repo with two packages, current `gatsby-remark-embedder` and a new `remark-embedder`. This later package would be the one with all functionality, distributed as remark plugin, gatsby plugin would just make use of it. This can be done maybe using [lerna](https://lerna.js.org/). An example of this structure I found is https://github.com/kevin940726/remark-codesandbox

The other options is to keep this as is, and just add a new `remark-embedder.js` file which should be used as a remark plugin whenever needed.

Wanted to ask first before working on this. Let me know!

### PR DESCRIPTION
**Pull request by kentcdodds**
**What**: feat: export a non-gatsby remark plugin

**Why**: Closes #136

**How**: Slightly refactor the main export to be generic and add two functions for the different APIs offered by remark and gatsby to call into the base function. Export both the remark and gatsby functions. Left the default export as the gatsby one so this could be published without breaking anything.

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,14 +64,10 @@ replace it with the proper embed-code.
...
   - [Twitter](#twitter)
   - [YouTube](#youtube)
-- [Options](#options-1)
+- [Options](#options)
   - [customTransformers](#customtransformers)
-    - [Properties](#properties)
-      - [`getHTML(url, options)`](#gethtmlurl-options)
-      - [`name`](#name)
-      - [`shouldTransform(url)`](#shouldtransformurl)
-    - [Example transformer](#example-transformer)
```
Comment (kentcdodds) [inline README.md:71]: This was removed when I used the `doctoc` functionality built-into kcd-scripts. Let me know if you'd like me to revert this change and add the remark plugin link by hand instead.

File: package.json [hunk 2]
```diff
@@ -58,9 +58,6 @@
     "pretty-quick": "^3.1.0",
     "remark": "^12.0.1"
   },
-  "peerDependencies": {
-    "gatsby": "^2.20.0"
-  },
```
Comment (kentcdodds) [inline package.json:63]: Technically you don't *have* to use gatsby to use this project.

I think the risk of someone mistakenly installing this without gatsby is extremely low, so I don't think there's much of a concern here.

**Comment by kentcdodds** [comment]
I just published this as [`remark-embedder@0.0.1`](https://www.npmjs.com/package/remark-embedder). I imagine several other things need to happen before this could *actually* be published (especially docs updates), but I thought I'd test it out and it worked well for me.

I'll give you publish rights to `remark-embedder` in case that's the direction you want to go.

**Comment by kentcdodds** [comment]
I found a bug and needed it fixed so I added it to this PR (let me know if you'd like me to move it to a different PR). I've published v0.0.2 of `remark-embedder` with this fix.

**Comment by wooorm** [comment]
What about making this a monorepo, which hosts both `remark-embedder` and `gatsby-remark-embedded`?

For turning a remark plugin into a gatsby plugin, this tiny wrapper can be used: https://github.com/kevin940726/to-gatsby-remark-plugin

**Comment by kentcdodds** [comment]
I think a monorepo might make sense for this repo. All the transformers could be individual packages and I think that would be really well-suited.

Unfortunately I don't have a lot of experience with that and don't have the time to figure out how to make our tooling and automation work well in that environment. So I think we'll push forward with the rename and go with the `remark-embedder/gatsby` approach. Perhaps in the future @MichaelDeBoey can transition to a monorepo if he wants.

Thanks for pointing me to that module. I'll give it a look.

File: src/__tests__/remark.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import remarkEmbedder from '../';
+
+test('works with remark directly', async () => {
+  const result = await remark()
+    .use(remarkEmbedder)
+    .process('https://www.youtube.com/watch?v=dQw4w9WgXcQ');
+
+  expect(result.toString()).toMatchInlineSnapshot(`
+    <iframe width="100%" height="315" src="https://www.youtube-nocookie.com/embed/dQw4w9WgXcQ?rel=0" frameBorder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen></iframe>
+  `);
```
Comment (kentcdodds) [inline src/__tests__/remark.js:11]: This assertion is failing:

```
 FAIL  src/__tests__/remark.js
  ✕ works with remark directly (16 ms)
  ✓ can pass options (2 ms)

  ● works with remark directly

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `works with remark directly 1`

    - Snapshot  - 1
    + Received  + 2

    - <iframe width="100%" height="315" src="https://www.youtube-nocookie.com/embed/dQw4w9WgXcQ?rel=0" frameBorder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen></iframe>
    + https://www.youtube.com/watch?v=dQw4w9WgXcQ
    +

       7 |     .process('https://www.youtube.com/watch?v=dQw4w9WgXcQ');
       8 | 
    >  9 |   expect(result.toString()).toMatchInlineSnapshot(`
         |                             ^
      10 |     <iframe width="100%" height="315" src="https://www.youtube-nocookie.com/embed/dQw4w9WgXcQ?rel=0" frameBorder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen></iframe>
      11 |   `);
      12 | });

      at Object.<anonymous> (src/__tests__/remark.js:9:29)

 › 1 snapshot failed.
Snapshot Summary
 › 1 snapshot failed from 1 test suite. Inspect your code changes or press `u` to update them.

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 passed, 2 total
Snapshots:   1 failed, 1 passed, 2 total
Time:        0.961 s, estimated 4 s
Ran all test suites matching /src\/__tests__\/remark\.js/i.
```

@wooorm, do you have any ideas? I think it may be related to the recent changes we made to use ASTs rather than a string of HTML.

Comment (kentcdodds) [reply src/__tests__/remark.js]: Nevermind! I just realized that I wasn't using `remark-html` which is responsible for actually taking the AST and turning it into a string of HTML. Once I added remark-html, the test passed.

**Comment by kentcdodds** [comment]
We're going to go a different direction with this package...
